integration/feature/init/src/MillInitMavenTests.scala

```diff
@@ -0,0 +1,221 @@
+            |[transport-native-kqueue] dropping classifier ${os.detected.classifier} for dependency io.netty:netty-tcnative:2.0.66.Final
+            |[handler] dropping classifier ${os.detected.classifier} for dependency io.netty:netty-tcnative:2.0.66.Final
+            |[example] dropping classifier ${os.detected.classifier} for dependency io.netty:netty-tcnative:2.0.66.Final
+            |[testsuite] dropping classifier ${os.detected.classifier} for dependency io.netty:netty-tcnative:2.0.66.Final
+            |[testsuite-shading] dropping classifier ${os.detected.classifier} for dependency io.netty:netty-tcnative:2.0.66.Final
+            |[transport-blockhound-tests] dropping classifier ${os.detected.classifier} for dependency io.netty:netty-tcnative:2.0.66.Final
+            |[microbench] dropping classifier ${os.detected.classifier} for dependency io.netty:netty-tcnative:2.0.66.Final
+            |""".stripMargin
+        ) ==> true
+        initRes.out.contains(
+          "generated 47 Mill build file(s)"
+        ) ==> true
+        initRes.isSuccess ==> true
+
+        val resolveRes = eval(("resolve", "_"))
+        Seq(
+          "all",
+          "dev-tools",
+          "common",
+          "buffer",
+          "codec",
+          "codec-dns",
+          "codec-haproxy",
+          "codec-http",
+          "codec-http2",
+          "codec-memcache",
+          "codec-mqtt",
+          "codec-redis",
+          "codec-smtp",
+          "codec-socks",
+          "codec-stomp",
+          "codec-xml",
+          "resolver",
+          "resolver-dns",
+          "resolver-dns-classes-macos",
+          "resolver-dns-native-macos",
+          "transport",
+          "transport-native-unix-common-tests",
+          "transport-native-unix-common",
+          "transport-classes-epoll",
+          "transport-native-epoll",
+          "transport-classes-kqueue",
+          "transport-native-kqueue",
+          "transport-rxtx",
+          "transport-sctp",
+          "transport-udt",
+          "handler",
+          "handler-proxy",
+          "handler-ssl-ocsp",
+          "example",
+          "testsuite",
+          "testsuite-autobahn",
+          "testsuite-http2",
+          "testsuite-osgi",
+          "testsuite-shading",
+          "testsuite-native",
+          "testsuite-native-image",
+          "testsuite-native-image-client",
+          "testsuite-native-image-client-runtime-init",
+          "transport-blockhound-tests",
+          "microbench",
+          "bom"
+        ).forall(resolveRes.out.contains) ==> true
+
+        // additional sources defined by build-helper-maven-plugin
+        val sourcesRes = eval(("show", "transport-native-epoll.sources"))
+        Seq(
+          s"$workspacePath/transport-native-epoll/src/main/java",
+          s"$workspacePath/transport-native-epoll/src/main/c"
+        ).forall(sourcesRes.out.contains) ==> true
```

Comment:
In this case, the additional source directory contains `C` sources. This is not useful.
The plugin supports additional directories for compile and test, sources and resources. Should sources be skipped?

Reply:
Do you know how the `src/main/c/` gets handled by Maven? For example, if we put a `Foo.java` file in there, will it get picked up by the Java compiler? 

If so we can keep it, if not maybe we can skip `sources` and just assume the defaults

Reply:
> Do you know how the src/main/c/ gets handled by Maven?

This is used to [publish](https://github.com/netty/netty/blob/64c52255c073d4a308d3dbb56a12d3cca7692c47/transport-native-epoll/pom.xml#L597) the `C` sources.

> For example, if we put a Foo.java file in there, will it get picked up by the Java compiler?

The compiler plugin is not used in this module since there are no Java sources.

I see 2 options for closing this:
- Limit `build-helper-maven-plugin` support to resources.
- Remove `build-helper-maven-plugin` support. Note that there are many ways to [configure additional sources](https://stackoverflow.com/questions/270445/maven-compile-with-multiple-src-directories).

Reply:
Went with option 2